Add availability property & default icon

mattsaxon opened the pull request:
Note, this change requires later version of pysonofflan (not committed at this time).

It also removes the now redundant call to update which is now invoked via a callback from pysonofflan. This means that switching only occurs once the switch has responded.

pyup-bot committed:
Update urllib3 from 1.24.1 to 1.24.2

pyup-bot committed:
Update cffi from 1.12.2 to 1.12.3

pyup-bot committed:
Update ruamel.yaml from 0.15.92 to 0.15.93

pyup-bot committed:
Update urllib3 from 1.24.2 to 1.25

pyup-bot committed:
Update ruamel.yaml from 0.15.93 to 0.15.94

mattsaxon committed:
First commit of full error handling and retry solution

mattsaxon committed:
remove debug

mattsaxon committed:
Add support for adding icons in configuration.yml

mattsaxon committed:
Update read.me to reference correct mdi

mattsaxon committed:
Create manifest.json

Although the docs say this is only needed for built in components, on upgrade to HA 91.3, without this file, I was getting errors that prevented the component loading

beveradb committed:
Update manifest.json

beveradb commented on manifest.json:
```diff
@@ -0,0 +1,8 @@
+{
+        "domain": "sonoff_lan_mode",
+        "name": "Sonoff LAN Mode",
```
I tweaked these slightly to fix typo and add dependency

beveradb commented on switch.py:
```diff
@@ -11,17 +11,19 @@
 _LOGGER = logging.getLogger('homeassistant.components.switch.sonoff_lan_mode')
 
 DEFAULT_NAME = 'Sonoff Switch'
+DEFAULT_ICON = 'mdi:flash'
```
Thanks, much better

beveradb commented:
LGTM, happy to merge once the current work on pysonofflan is complete and we've released a new version to pypi etc.

beveradb committed:
Delete requirements_dev.txt

I don't believe there is any reason for requirements_dev.txt to exist in this repository - this isn't actually a python library or something which is usable without being run as part of a home assistant installation, so any requirements*.txt file here will never be used?

Happy to be corrected if I've missed something obvious here, but pretty sure that file still existing here was just a leftover from before I implemented this properly and when it was still a hacky standalone script 😉

mattsaxon committed:
Updated pysonofflan dependency

mattsaxon committed:
Merge branch 'master' of https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

mattsaxon commented:
Now you've updated the PySonoffLan component, I've updated the requirement line in switch.py and I think this is ready to merge.

beveradb reviewed (approved):
LGTM
